sdk/search/Azure.Search.Documents/tests/Utilities/SearchRecordedTestSanitizer.cs

```diff
@@ -29,5 +61,26 @@ public override void SanitizeHeaders(IDictionary<string, string[]> headers)
             }
             base.SanitizeHeaders(headers);
         }
+
+        /// <summary>
+        /// Redact sensitive variables.
+        /// </summary>
+        /// <param name="variableName">The name of the variable.</param>
+        /// <param name="environmentVariableValue">The value of the variable.</param>
+        /// <returns>The sanitized variable value.</returns>
+        public override string SanitizeVariable(string variableName, string environmentVariableValue)
+        {
+            if (SearchResources.StorageAccountKeyVariableName.Equals(variableName, StringComparison.OrdinalIgnoreCase))
+            {
+                // Assumes the secret content is destined to appear in JSON, for which certain common characters in account keys are escaped.
+                // See https://github.com/dotnet/runtime/blob/8640eed0/src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Escaping.cs
+                string encoded = JavaScriptEncoder.Default.Encode(environmentVariableValue);
+                _secrets.Add(encoded);
+
+                return SanitizeValue;
```

Comment:
So this replaces the entire connection string with "SANITIZED"?  I suppose if we're not actually uploading anything (and consequently instrumenting our BlobContainerClient, etc.) then it's fine.

Reply:
No, just the AccountKey. See `SanitizeBodyText` above. I suppose it could sanitize the whole thing, but the rest doesn't contain anything not already exposed (and okay to do so), but I don't see any benefit to push another commit.

Reply:
Oh, sorry, I thought we had a connection string.  Makes sense now.